src/sparsezoo/utils/onnx.py

```diff
@@ -42,6 +50,100 @@
 ]
 
 
+def save_onnx(
+    model: ModelProto, model_path: str, external_data_file: Optional[str] = None
+) -> bool:
+    """
+    Save model to the given path.
+
+    If the model's size is larger than the maximum protobuf size:
+        -   it will be saved with external data
+    If the model's size is smaller than the maximum protobuf size:
+        -   and the user nevertheless specifies 'external_data_file',
+            the model will be saved with external data.
+
+    :param model: The model to save.
+    :param model_path: The path to save the model to.
+    :param external_data_file: The optional name save the external data to.
+    :return True if the model was saved with external data, False otherwise.
+    """
+    if external_data_file is not None:
+        _LOGGER.info(f"Saving with external data: {external_data_file}")
+        onnx.save(
+            model,
+            model_path,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location=external_data_file,
+        )
+        return True
+
+    if model.ByteSize() > onnx.checker.MAXIMUM_PROTOBUF:
+        _LOGGER.info(
+            "The ONNX model is too large to be saved as a single protobuf."
+            "Saving with external data... "
+        )
+
+        _LOGGER.info(f"Saving with external data: {external_data_file}")
+        onnx.save(
+            model,
+            model_path,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+        )
+        return True
+
+    onnx.save(model, model_path)
+    return False
+
+
+def validate_onnx(model: Union[str, ModelProto]):
+    """
+    Validate that a file at a given path is a valid ONNX model.
+    Raises a ValueError if not a valid ONNX model.
+
+    :param model: the model proto or path to the model
+        ONNX file to check for validation
+    """
+    try:
+        onnx_model = load_model(model)
+        if onnx_model.ByteSize() > onnx.checker.MAXIMUM_PROTOBUF:
+            if isinstance(model, str):
+                onnx.checker.check_model(model)
+            else:
+                _LOGGER.warning(
+                    "Attempting to validate an in-memory ONNX model with "
+                    f"size > {onnx.checker.MAXIMUM_PROTOBUF} bytes."
+                    "`validate_onnx` skipped, as large ONNX models cannot "
+                    "be validated in-memory. To validate this model, save "
+                    "it to disk and call `validate_onnx` on the file path."
+                )
+            return
+        onnx.checker.check_model(onnx_model)
+        if not onnx_model.opset_import:
+            raise ValueError("could not parse opset_import")
```

Comment:
What is this guarding against?

Reply:
This was there before in the original function (https://github.com/neuralmagic/sparseml/blob/main/src/sparseml/onnx/utils/helpers.py#L98), not really sure what's the function here.
I think Mark is the one who wrote it and added it in this commit: https://github.com/neuralmagic/sparseml/commit/4c1ae34616db21cb594668fb82c588b9a43d7c94

Reply:
Ah ok, I guess it's just checking that the model has an opset number